### PR TITLE
Build: move to 'dependencies' the ones required for running a build

### PR DESCRIPTION
**Pull request by alfonsomthd**

### CODE REVIEW

File: .github/workflows/frontend-build.yml [hunk 1]
```diff
@@ -18,11 +18,12 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
           cache: 'yarn'
-      - run: yarn --frozen-lockfile
+      - run: yarn install --prod --frozen-lockfile
       - run: yarn build
+      - run: yarn build-mco
+      - run: yarn install # Install dev. deps.
```
Comment (GowthamShanmugam) [inline .github/workflows/frontend-build.yml:24]: is first yarn install for prod dependency and second for dev? do we need both?

Comment (alfonsomthd) [reply .github/workflows/frontend-build.yml:24]: @GowthamShanmugam yes, the first `yarn install --prod` is to ensure that there is no missing dep. required for a build, the second one installs the lint libraries as the enxt step is to run the lint.

Comment (GowthamShanmugam) [reply .github/workflows/frontend-build.yml:24]: ack No further comments from my side its LGTM

Comment (GowthamShanmugam) [reply .github/workflows/frontend-build.yml:24]: cc: @bipuladh @SanjalKatiyar

File: package.json [hunk 2]
```diff
@@ -60,8 +63,15 @@
     "@patternfly/react-tokens": "^4.12.0",
     "@patternfly/react-topology": "^4.90.38",
     "@types/react-dnd-html5-backend": "^3.0.2",
+    "@types/lodash-es": "^4.17.4",
```
Comment (bipuladh) [inline package.json:66]: not required

Comment (alfonsomthd) [reply package.json:66]: @bipuladh I re-tested it previously delting deleting `node_modules`and the build fails without it:
```
ERROR in ../../packages/odf/utils/mcg.ts:213:20
TS2538: Type 'unknown' cannot be used as an index type.
    211 |   let type: BC_PROVIDERS = null;
    212 |   Object.entries(NS_PROVIDERS_NOOBAA_MAP).forEach(([k, v]) => {
  > 213 |     if (ns?.spec?.[v]) {
        |                    ^
    214 |       type = k as BC_PROVIDERS;
    215 |     }
    216 |   });

ERROR in ../../packages/shared/charts/utils.ts:88:13
TS2349: This expression is not callable.
  Not all constituents of type 'string | ((result: PrometheusResult, index: number) => string)' are callable.
    Type 'string' has no call signatures.
    86 |         y: threshold ? threshold : yMutator?.(y) ?? defaultYMutator(y),
    87 |         description: _.isFunction(description)
  > 88 |           ? description(r, index)
       |             ^^^^^^^^^^^
    89 |           : description,
    90 |         symbol,
    91 |       } as DataPoint<Date>;

ERROR in ../../packages/shared/generic/ButtonBar.tsx:10:29
TS2339: Property 'type' does not exist on type 'string | number | ReactElement<any, string | JSXElementConstructor<any>>'.
  Property 'type' does not exist on type 'string'.
     8 | const injectDisabled = (children: React.ReactChild, disabled) => {
     9 |   return React.Children.map(children, (c) => {
  > 10 |     if (!_.isObject(c) || c.type !== 'button') {
       |                             ^^^^
    11 |       return c;
    12 |     }
    13 |

ERROR in ../../packages/shared/generic/ButtonBar.tsx:14:31
TS2769: No overload matches this call.
  The last overload gave the following error.
    Argument of type 'string | number | ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to parameter of type 'ReactElement<any, string | JSXElementConstructor<any>>'.
      Type 'string' is not assignable to type 'ReactElement<any, string | JSXElementConstructor<any>>'.
    12 |     }
    13 |
  > 14 |     return React.cloneElement(c, { disabled: c.props.disabled || disabled });
       |                               ^
    15 |   });
    16 | };
    17 |

ERROR in ../../packages/shared/generic/ButtonBar.tsx:14:48
TS2339: Property 'props' does not exist on type 'string | number | ReactElement<any, string | JSXElementConstructor<any>>'.
  Property 'props' does not exist on type 'string'.
    12 |     }
    13 |
  > 14 |     return React.cloneElement(c, { disabled: c.props.disabled || disabled });
       |                                                ^^^^^
    15 |   });
    16 | };
    17 |

webpack 5.74.0 compiled with 5 errors in 55702 ms

```

**Comment by SanjalKatiyar** [comment]
> /approve

@bipuladh for "lgtm"

**Comment by SanjalKatiyar** [comment]
@alfonsomthd this PR is planned for 4.13 right ?

**Comment by alfonsomthd** [comment]
> @alfonsomthd this PR is planned for 4.13 right ?

Yes, I plan to backport it.

**Pull request merged**